wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/spi/RuntimeInfrastructure.java

```diff
@@ -79,6 +82,20 @@ public EventService getEventService() {
     throw new UnsupportedOperationException("The implementation does not track runtimes");
   }
 
+  /**
+   * This oddly specific method tries to find discrepancies between the workspace activity records
+   * and the true state of the workspaces as tracked by the runtime.
+   *
+   * <p>Because tracking of runtimes is specific to the backing infrastructure the workspace
+   * activity tracking, which is infrastructure agnostic, needs to ask the infrastructure for this
+   * information.
+   *
+   * @return a map where keys are workspace IDs and values are true workspace statuses that are not
+   *     correctly reflected in the workspace activity records
+   */
```

Comment:
why don't we put this in workspace activity DAO?
@sleshchenko @metlos wdyt?

Reply:
Notice that the impl is infrastructure specific - i.e. we have different infrastructures and for each, the implementation can differ (it does not atm, but from how the code is laid out, it can). So we cannot put this into activity dao, because that would mix concerns. `che_k8s_runtime` is a table "private" to the kubernetes runtime and IMHO should not be accessed by other modules.

Reply:
After discussion, we'll not use this implementation.